s3/main.tf

```diff
@@ -2,8 +2,15 @@ locals {
   log_bucket_count = var.create_log_bucket ? 1 : 0
   log_bucket_name  = "${var.bucket_name}-logs"
 }
+
+resource "random_string" "bucket_suffix" {
+  length  = 8
+  special = false
+  numeric = false
+}
+
 resource "aws_s3_bucket" "bucket" {
-  bucket = var.bucket_name
+  bucket = "${var.bucket_name}-${random_string.bucket_suffix.result}"
```

Comment:
How difficult is string manipulation in Terraform, could we truncate if too long before adding the random bit?

Reply:
I just realised what you're saying, truncate the bucket name, then add the prefix. There's a substr function that should work here.

Reply:
Wouldn't it be better to throw an error or something  instead? Or enforce that the bucket name is 54 characters long?

Reply:
I don't think it should be a blocker as a bucket will still be created in this case. I'm copying the CloudFormation default of truncate+suffix.